modules/zip/src/parse-zip/cd-file-header.ts

```diff
@@ -61,30 +63,62 @@ export const parseZipCDFileHeader = async (
   const extraOffset = headerOffset + CD_FILE_NAME_OFFSET + BigInt(fileNameLength);
   const oldFormatOffset = await file.getUint32(headerOffset + CD_LOCAL_HEADER_OFFSET_OFFSET);
 
-  let fileDataOffset = BigInt(oldFormatOffset);
-  let offsetInZip64Data = 4n;
+  const localHeaderOffset = BigInt(oldFormatOffset);
+  const extraField = new Uint8Array(
+    await file.slice(extraOffset, extraOffset + BigInt(extraFieldLength))
+  );
   // looking for info that might be also be in zip64 extra field
+
+  const zip64data = {
+    uncompressedSize,
+    compressedSize,
+    localHeaderOffset,
+    startDisk
+  };
+  // We define fields that should be in zip64 data
+  const zip64dataList: {length: number; name: string}[] = [];
   if (uncompressedSize === BigInt(0xffffffff)) {
-    uncompressedSize = await file.getBigUint64(extraOffset + offsetInZip64Data);
-    offsetInZip64Data += 8n;
+    zip64dataList.push({name: 'uncompressedSize', length: 8});
   }
   if (compressedSize === BigInt(0xffffffff)) {
-    compressedSize = await file.getBigUint64(extraOffset + offsetInZip64Data);
-    offsetInZip64Data += 8n;
+    zip64dataList.push({name: 'compressedSize', length: 8});
```

Comment:
This code is very long and is getting hard to follow. Can we break into well-named self documenting sub-functions?